android/jni/com/mapswithme/maps/editor/Editor.cpp

```diff
@@ -274,4 +287,46 @@ Java_com_mapswithme_maps_editor_Editor_nativeGetNewFeatureCategories(JNIEnv * en
 
   return jCategories;
 }
+
+JNIEXPORT jobjectArray JNICALL
+Java_com_mapswithme_maps_editor_Editor_nativeGetCuisines(JNIEnv * env, jclass clazz)
+{
+  osm::TAllCuisines const & cuisines = osm::Cuisines::Instance().AllSupportedCuisines();
+  vector<string> keys(cuisines.size());
```

Comment:
Окей, поправлю
> On Mar 22, 2016, at 11:14 AM, Alex Zolotarev <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/editor/Editor.cpp <https://github.com/mapsme/omim/pull/2391#discussion_r56949051>:
> 
> > @@ -274,4 +287,46 @@ Java_com_mapswithme_maps_editor_Editor_nativeGetNewFeatureCategories(JNIEnv * en
> >  
> >    return jCategories;
> >  }
> > +
> > +JNIEXPORT jobjectArray JNICALL
> > +Java_com_mapswithme_maps_editor_Editor_nativeGetCuisines(JNIEnv * env, jclass clazz)
> > +{
> > +  osm::TAllCuisines const & cuisines = osm::Cuisines::Instance().AllSupportedCuisines();
> > +  vector<string> keys(cuisines.size());
> Мелочь: такой подход создаст много пустых строк в векторе, а дальше в цикле их переприсвоит.
> 
> Можно слегка эффективнее: создать пустой вектор, сделать ему reserve (не resize!), и дальше через push_back.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2391/files/f423aeb4c2632729a07c8764af99345e0b13e77b#r56949051>